frontend/src/pages/Admin/TrainingReports.js

```diff
@@ -0,0 +1,181 @@
+
+    // Clear error and success messages.
+    setError('');
+    setSuccess('');
+    setInfo('');
+
+    // Verify that the file is a CSV.
+    if (files.length === 1 && files[0].type !== 'text/csv') {
+      setError('Please upload a CSV file.');
+      setUploadDisabled(true);
+      e.target.value = null;
+      return;
+    }
+
+    const reader = new FileReader();
+    reader.onload = (event) => {
+      // Load the csv.
+      const csv = event.target.result;
+
+      // Replace all instances of LF that are not followed by a CR with a space.
+      const cleanCsv = csv.replace(/\r(?!\n)/g, ' ');
+
+      // Get all headers.
+      const headers = cleanCsv.split('\r\n')[0].split(',');
+
+      // Validate columns.
+      const requiredHeaders = requiredCsvHeaders.filter((header) => !headers.includes(header));
+      const invalidHeaders = headers.filter((header) => !validCsvHeaders.includes(header));
+      if (requiredHeaders.length > 0) {
+        setError(`Required headers missing: ${requiredHeaders.join(', ')}`);
+        setUploadDisabled(true);
+        e.target.value = null;
+        return;
+      }
+      if (invalidHeaders.length > 0) {
+        setError(`Invalid headers found: ${invalidHeaders.join(', ')}`);
+        setUploadDisabled(true);
+        e.target.value = null;
+        return;
+      }
+
+      // Get a distinct count of 'Event ID' column's.
+      const eventIds = cleanCsv.split('\r\n').map((row) => row.split(',')[1]).filter((id) => id !== 'Event ID' && id !== undefined);
+      const distinctEventIds = [...new Set(eventIds)];
+
+      // Check if we have duplicate 'Event ID' values.
+      if (distinctEventIds.length !== eventIds.length) {
+        setError('Duplicate Event IDs found. Please correct and try again.');
+        setUploadDisabled(true);
+        e.target.value = null;
+        return;
+      }
+
+      // Display count of distinct ids.
+      setInfo(`${distinctEventIds.length} events will be imported.`);
+
+      // Clear errors and enable import button.
+      setError('');
+      setUploadDisabled(false);
+    };
+
+    // call reader with the file.
+    reader.readAsText(files[0]);
+  };
+  return (
+    <>
+      <Container paddingX={1} paddingY={1} className="smart-hub--overflow-auto">
+        <div>
+          <h2>Training Report Import</h2>
+          {(success && !error) && (
+            <Alert type="success" className="margin-bottom-1 maxw-mobile-lg" noIcon>
+              {success}
+            </Alert>
+          )}
+          {error && (
+            <Alert type="error" className="margin-bottom-1 maxw-mobile-lg" noIcon>
+              {error}
+            </Alert>
+          )}
+          {info && (
+          <Alert type="info" className="margin-bottom-1 maxw-mobile-lg" noIcon>
+            {info}
+          </Alert>
+          )}
+          <div className="display-flex">
+            <FormGroup>
+              <Label htmlFor="file-input-single">Input accepts a single file</Label>
```

Comment:
htmlFor has to match the id on the FileInput

Reply:
Done.